### DISCUSSION TITLE
블로그 스케일 업 관련 문제

### DISCUSSION THREAD

**Discussion by SongDerrick** [morethanmin/morethan-log#170] [Ideas]
안녕하세요! 블로그 잘 사용하고 있습니다.

시간 되시면 놀러오세요... -> https://doyoonsong.vercel.app/

다만 블로그 스케일 업 과정에서 문제가 있는데...
이제 블로그 포스트 50여개가 넘어가다 보니, vercel hobby 수준에서 이 프로젝트에 사용된 로직으로는... 로드를 감당하기 어려워지고 있습니다.

작성하신 코드를 보니, 메인 페이지에서도 각각의 포스트의 전체 데이터가 json 형식으로 클라이언트로 넘어오고 있는 것으로 보입니다.
여기서 시간이 너무 오래걸리다 보니 10초 제한이 있는 vercel hobby에서 감당하지 못하고 블로그가 업데이트 되지 않고 400에러가 발생하고 있습니다.

블로그를 스케일 업 하기 위해서는, 

1. 관련 메타 데이터만 받아와서 메인 페이지에 나열하도록 하고, 각 페이지에 접속했을 때 나머지 컨텐츠 데이터를 json 파일로 받는 방식으로 변경하는 방법.

2. 모든 포스트에 대한 데이터를 수신하지 않고, 20개 혹은 10개 단위로 포스트 데이터를 일부분만 수신하는 방법등이 필요한 것 같습니다..

관련해서 개선을 해보려고 하는데, 혹시 다른 아이디어가 있으신가요?  저는 2번 방식으로 문제 해결을 해보려고 합니다!

**Comment by morethanmin** [comment]
안녕하세요! 우선 관심 가져주셔서 감사합니다.

현재 `notion-client` 라이브러리를 사용해 노션 페이지 정보를 한번에 읽어오다보니 생겨난 문제같습니다. 😢 
말씀 주신 방법으로 개선이 된다면 좋을 것 같은데, 혹시 생각해보신 구현 방안이 있으신가요?

제가 생각해본 방안으로는 notion official sdk를 통해 database를 조회하고 렌더링 해주는 방식이 있을 것 같습니다

**Reply by SongDerrick** [reply]
네 현재 두번째 방법으로 구현을 해 봤습니다! 
20개의 포스트만 화면에 보여지도록 포스트 제공 갯수를 제한하는 접근으로 구현했습니다. 
그리고 초과되는 포스트는 1,2,3.. 이런식으로 다음 페이지 번호를 클릭하면 확인 가능하도록 해놓았습니다...
맨 아래까지 스크롤 해보시면 확인 가능합니다! ->  https://doyoonsong.vercel.app/

스크롤이 내려가면서 뷰포트에 담기는 포스트가 생길 때마다 해당 포스트의 json파일이 수신되는 것으로 보여서, 
이렇게 구현을 해봤더니 일단은 문제 없이 잘 작동하는 것으로 보입니다!
다만 자잘한 버그가 있어서 버그 픽스한 후에 이슈 탭에 올리겠습니다!

**Comment by khakhiD** [comment]
데릭님 덕분에 블로그 참고하여 페이지네이션 적용했습니다.
혹시 말씀하신 자잘한 버그가 무엇인지 알 수 있을까요?
